monai/visualize/utils.py

```diff
@@ -0,0 +1,121 @@
+            A list of channel-first (C, H[, W, D]) arrays can also be passed in,
+            in which case they will be displayed as a padded and stacked volume.
+        fig: matplotlib figure to use. If None, a new figure will be created.
+        title: title of the figure.
+        figsize: size of the figure.
+        frames_per_row: number of frames to display in each row. If None, sqrt(firstdim) will be used.
+        vmin: `vmin` for the matplotlib `imshow`.
+        vmax: `vmax` for the matplotlib `imshow`.
+        every_n: factor to subsample the frames so that only every n-th frame is displayed.
+        interpolation: interpolation to use for the matplotlib `matshow`.
+        show: if True, show the figure.
+        fill_value: value to use for the empty part of the grid.
+        kwargs: additional keyword arguments to matplotlib `matshow` and `imshow`.
+
+    See Also:
+        - https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.imshow.html
+        - https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.matshow.html
+
+    Example:
+
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from monai.visualize import matshow3d
+        # figure 3D volume
+        >>> volume = np.random.rand(10, 10, 10)
+        >>> fig = plt.figure()
+        >>> matshow3d(volume, fig=fig, title="3D Volume")
+        >>> plt.show()
+        # figure a list of 3D volumes
+        >>> volumes = [np.random.rand(1, 10, 10, 10), np.random.rand(1, 10, 10, 10)]
+        >>> fig = plt.figure()
+        >>> matshow3d(volumes, fig=fig, title="List of Volumes")
+        >>> plt.show()
+
+    """
+    vol: np.ndarray = convert_data_type(data=volume, output_type=np.ndarray)[0]  # type: ignore
+    if isinstance(vol, (list, tuple)):
+        # a sequence of channel-first volumes
+        if not isinstance(vol[0], np.ndarray):
+            raise ValueError("volume must be a list of arrays.")
+        pad_size = np.max(np.asarray([v.shape for v in vol]), axis=0)
+        pad = SpatialPad(pad_size[1:])  # assuming channel-first for item in vol
+        vol = np.concatenate([pad(v) for v in vol], axis=0)
+    else:
+        while len(vol.shape) < 3:
+            vol = np.expand_dims(vol, 0)  # so that we display 1d and 2d as well
+    if len(vol.shape) > 3:
+        vol = vol.reshape((-1, vol.shape[-2], vol.shape[-1]))
+    vmin = np.nanmin(vol) if vmin is None else vmin
+    vmax = np.nanmax(vol) if vmax is None else vmax
+
+    # making grid of every_n-th frame
+    vol = vol[:: max(every_n, 1)]
+    if not frames_per_row:
+        frames_per_row = int(np.ceil(np.sqrt(len(vol))))
+    frames_per_row = max(min(len(vol), frames_per_row), 1)
+    height, width = vol.shape[-2:]
+    n_rows = int(np.ceil(len(vol) / frames_per_row))
+    im = np.full(shape=(height * n_rows, width * frames_per_row), fill_value=fill_value, dtype=vol.dtype)
```

Comment:
I had in the past use something like the following to arrange a stack of images into a tile arrangement that might be a better solution:

```python
def tile_stack(stack,cols,rows=1):
    b=len(stack)
    stack=stack[:b-b%cols]
    rows=min(rows,len(stack)//cols)
    out=[]
    
    for r in range(rows):
        out.append([stack[r*cols+c] for c in range(cols)])
        
    return np.block(out)
```

Reply:
Many thanks, the method is nice!